connector/src/main/scala/com/google/cloud/spark/bigquery/direct/DirectBigQueryRelation.scala

```diff
@@ -124,11 +124,13 @@ private[bigquery] class DirectBigQueryRelation(
 
       val maxNumPartitionsRequested = getMaxNumPartitionsRequested(actualTableDefinition)
 
+      val readDataFormat = options.readDataFormat
+
       try {
         val session = client.createReadSession(
           CreateReadSessionRequest.newBuilder()
             .setParent(s"projects/${options.parentProject}")
-            .setFormat(DataFormat.AVRO)
+            .setFormat(readDataFormat)
```

Comment:
Make sure to mark `Or` as unhandled filter for Arrow in `DirectBigQueryReltion.isHandled()`

Reply:
Done